Paco-Server/src/com/google/sampling/experiential/server/ACLHelper.java

```diff
@@ -52,17 +58,26 @@ public static String getModifiedQueryBasedOnACL(Select selectSql, String loggedI
     if (adminExperimentsinDB != null && !adminExperimentsinDB.containsAll(userSpecifiedExpIdValues)) {
       adminOnAllExperiments = false;
     }
-
-    if (!adminOnAllExperiments && !onlyQueryingOwnData) {
+    
+    // if the user is an admin on atleast one experiment, then check if he has full access/ACL on all expts in query
+    // and also if he is not querying just his data
+    if (!onlyAParticipant && !adminOnAllExperiments && !onlyQueryingOwnData) {
       throw new Exception(ErrorMessages.UNAUTHORIZED_ACCESS_MIXED_ACL.getDescription());
     }
-
-    if ((adminExperimentsinDB != null && adminExperimentsinDB.size() == 0) && !onlyQueryingOwnData) {
+    
+    // if logged in user has never been an admin of any experiment  
```

Comment:
Ditto from above: Just to clarify the language, it seems like it should say "is the user only a participant". The "has never been an admin" phrase seems confusing because we don't keep a history of if they have ever been an admin - it only matters if they are presently an admin.